cmd/kpod/common.go

```diff
@@ -133,3 +136,19 @@ func validateFlags(c *cli.Context, flags []cli.Flag) error {
 	}
 	return nil
 }
+
+// getCachedCreds returns the DockerAuthConfig of a registry if stored in the authfile
+func getCachedCreds(registry string, sc *types.SystemContext) (*types.DockerAuthConfig, error) {
+	if registry == "" {
+		return nil, nil
+	}
+	user, pass, err := config.GetAuthentication(sc, registry)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting credentials from authfile for %q", registry)
+	}
+	creds, err := common.ParseRegistryCreds(user + ":" + pass)
```

Comment:
This is…inelegant. Perhaps `ParseRegistryCreds` could be split into a function that does parsing, and a function that does the configuration.